src/app/tools/inks.h

```diff
@@ -366,7 +366,8 @@ class EraserInk : public BaseInk {
       }
       case ReplaceFgWithBg:
         loop->setPrimaryColor(loop->getFgColor());
-        loop->setSecondaryColor(loop->getBgColor());
+        loop->setSecondaryColor(loop->getBgColor() |
+          (loop->getLayer()->isBackground() ? ColorAMask : 0));
```

Comment:
This works only for RGB images, but here `getBgColor()` and `setSecondaryColor()` could be handling an index or a grayscale color (the bug is reproducible in a grayscale image even with this patch).